explainer.md

```diff
@@ -42,43 +42,47 @@ VR provides an interesting canvas for artists looking to explore the possibiliti
+`navigator.vr.requestDevice` returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that resolves to a `VRDevice` if one is available. In no `VRDevice` is available it will resolve to null. The promise will be rejected if an error occurs, such as the the page not having the appropriate permissions to access VR capabilities.
+
+A `VRDevice` represents a physical unit of VR hardware that can present imagery to the user somehow, referred to here as a "VR hardware device". On desktop clients this will usually be a headset peripheral; on mobile clients it may represent the mobile device itself in conjunction with a viewer harness (e.g., Google Cardboard/Daydream or Samsung GearVR). It may also represent devices without stereo-presentation capabilities but with more advanced tracking, such as ARCore/ARKit-compatible devices.
 
 ```js
-let vrDevice = null;
+function checkForVR() {
+  navigator.vr.requestDevice().then(device => {
+    if (device) {
+      onVRAvailable(device);
+    }
+  }, err => {
+    // An error occurred while requesting a VRDevice.
+    console.error('Unable to request a VR device.');
+  });
+}
+```
 
-navigator.vr.getDevices().then(devices => {
-  if (devices.length > 0) {
-    // Use the first device in the array if one is available. If multiple
-    // devices are present, you may want to provide the user a way of choosing
-    // which device to use.
-    vrDevice = devices[0];
-    onVRAvailable();
-  } else {
-    // Could not find any VR hardware connected.
-  }
-}, err => {
-  // An error occurred querying VR hardware. May be the result of blocked
-  // permissions by a parent frame.
-});
+Future revisions of the API may add filter criteria to `navigator.vr.requestDevice`.
+
+> **Non-normative Note:** If there are multiple VR devices available, the UA will need to pick which one to return. The UA is allowed to use any criteria it wishes to select which device is returned, including settings UI that allows users to manage device priority. Calling `navigator.vr.requestDevice` should not trigger device selection UI, however, as this would cause many sites to display VR-specific dialogs early in the document lifecycle without and user activation.
+
+It's possible that even if no VR device is available initially one may become available while the application is running, or that a previously available device becomes unavailable. This will be most common with PC peripherals that can be connected at disconnected at any time. To respond to these changes pages can listen to the `devicechange` event on the `navigator.vr` object.
+
+```js
+navigator.vr.addEventListener('devicechange', checkForVR);
```

Comment:
will `devicechange` always fire when a document loads and a VR device is detected? or in the presence of an event listener on `navigator.vr`? what's the trigger?

if so, when? in relation to `DOMContentLoaded`, `load`, etc.?